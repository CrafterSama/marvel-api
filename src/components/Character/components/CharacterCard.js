import React from 'react'
import PropTypes from 'prop-types'
import { Card, Overlayer, CardInfo, CardName } from '../../../styles'
import { getCharactersComics } from '../../../context/actions'
import { useStateValue } from '../../../context'

const CharacterCard = ({ data, showCharacterInfo, setData  }) => {

  const  [state, dispatch] = useStateValue()

  const showModal = (event) => {
    event.preventDefault()
    setData(data)
    getCharactersComics(data.id, state, dispatch)
    showCharacterInfo()
  }

  return (
    <Card background={data.thumbnail}>
      <Overlayer />
      <CardInfo onClick={showModal}>
        <CardName>
          <span>{data.name}</span>
        </CardName>
        <div className='character-favorite'></div>
      </CardInfo>
    </Card>
  );
}

CharacterCard.propTypes = {
  data: PropTypes.object.isRequired
}

export default CharacterCard
