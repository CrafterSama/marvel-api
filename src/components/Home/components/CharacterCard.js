import React from 'react'
import PropTypes from 'prop-types'
import { Card, Overlayer, CharacterInfo, CharacterName } from '../../../styles'

const CharacterCard = ({ data }) => {
  return (
    <Card background={data.thumbnail}>
      <Overlayer />
      <CharacterInfo>
        <CharacterName>
          <span>{data.name}</span>
        </CharacterName>
        <div className="character-favorite"></div>
      </CharacterInfo>
    </Card>
  )
}

CharacterCard.propTypes = {
  data: PropTypes.object.isRequired
}

export default CharacterCard
