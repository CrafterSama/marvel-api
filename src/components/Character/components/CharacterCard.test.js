import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import CharacterCard from './CharacterCard';

configure({ adapter: new Adapter() });

describe('Test for <CharacterCard /> Component', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(<CharacterCard data={{ data: {} }} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
