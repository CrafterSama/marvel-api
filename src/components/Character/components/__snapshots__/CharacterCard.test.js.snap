// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <CharacterCard /> Component matches snapshot 1`] = `
<styles__Card>
  <styles__Overlayer />
  <styles__CardInfo
    onClick={[Function]}
  >
    <styles__CardName>
      <span />
    </styles__CardName>
    <div
      className="character-favorite"
    />
  </styles__CardInfo>
</styles__Card>
`;
